### ISSUE TITLE
Prompt when interrupted still hides the cursor

### ISSUE THREAD

**Issue by kodematthieu** [mitsuhiko/dialoguer#188]
For example you call `Confirm::new("Do you want to continue").interact_opt()` and on the terminal you enter `CTRL+C`, the cursor will remain hidden. This is acceptable though because the program terminates and does not allow you to exit cleanly.

But for example you have `signal-hook` to handle signals, the program will continue with the cursor still hidden. I think this should be handled by the crate as it was the one that hid the cursor

**Comment by kraktus** [comment]
I agree this should be handled by the crate, at least as an opt-in option for the user

**Comment by untitaker** [comment]
This is an exact duplicate of https://github.com/mitsuhiko/dialoguer/issues/77, and the workaround there is insufficient. To my knowledge at least SIGINT and SIGPIPE need to be handled this way in order to avoid the cursor from being permanently broken.

I understand that dialoguer is not in an optimal position to solve this problem without potentially interfering with user-defined signal handlers. Perhaps we can have an option to disable cursor-hiding altogether instead?
